ext/opcache/jit/zend_jit.c

```diff
@@ -4044,6 +4044,17 @@ ZEND_EXT_API void zend_jit_protect(void)
 #endif
 }
 
+static void zend_jit_init_non_hybrid_handlers(void)
+{
+	zend_jit_runtime_jit_handler = (const void*)zend_runtime_jit;
+	zend_jit_profile_jit_handler = (const void*)zend_jit_profile_helper;
+	zend_jit_func_hot_counter_handler = (const void*)zend_jit_func_counter_helper;
+	zend_jit_loop_hot_counter_handler = (const void*)zend_jit_loop_counter_helper;
+	zend_jit_func_trace_counter_handler = (const void*)zend_jit_func_trace_helper;
+	zend_jit_ret_trace_counter_handler = (const void*)zend_jit_ret_trace_helper;
+	zend_jit_loop_trace_counter_handler = (const void*)zend_jit_loop_trace_helper;
```

Comment:
I think it would be cleaner to do this for the hybrid case as well. May not be necessary, but it makes it more consistent.

Reply:
Heck, totally forgot about this PR. Sorry!

I agree that this would be more consistent, but I'm afraid that `dasm_labels` isn't properly initialized when reattaching. There is code in `zend_jit_make_stubs()` which seems to do that:
https://github.com/php/php-src/blob/5fb5a739e2470a89b8620ab9d4afb265e36fa3ab/ext/opcache/jit/zend_jit.c#L4052-L4063
I doubt that we can call `zend_jit_make_stubs()` from this code path (i.e. when re-attaching).

Reply:
The stub init can be separated from the assignments, no?

Reply:
My understanding was that this whole if/else should be extracted. https://github.com/php/php-src/pull/7208/files#diff-9faa6c4e1812bb88ee722439bfa2d0b48c8034871c2a625e00d14e6dbfbb835cR4076

Reply:
@cmb When reattaching, dasm_labels is restored here: https://github.com/php/php-src/blob/5fb5a739e2470a89b8620ab9d4afb265e36fa3ab/ext/opcache/jit/zend_jit.c#L4355 As @dktapps suggested, we can extract this whole if/else and then call it once in zend_jit_make_stubs and once below the dasm_labels restore.

Reply:
Ah, right. Thanks!